source/common/http/conn_manager_impl.cc

```diff
@@ -423,23 +423,36 @@ void ConnectionManagerImpl::onEvent(Network::ConnectionEvent event) {
 
   if (event == Network::ConnectionEvent::RemoteClose ||
       event == Network::ConnectionEvent::LocalClose) {
-    if (connection_idle_timer_) {
-      connection_idle_timer_->disableTimer();
-      connection_idle_timer_.reset();
-    }
+    // TODO(mattklein123): It is technically possible that something outside of the filter causes
+    // a local connection close, so we still guard against that here. A better solution would be to
+    // have some type of "pre-close" callback that we could hook for cleanup that would get called
+    // regardless of where local close is invoked from. Note that this will cause this method to
+    // get called twice in the common local close cases, but the method protects against that.
```

Comment:
wait, which method gets called twice?  I thought doConnectionClose would.  If so can you clarify because it reads like onEvent would

Reply:
Sorry yes it's doConnectionClose, will clarify.